test/unit/concerns/smart_proxy_reference_test.rb

```diff
@@ -0,0 +1,17 @@
+require 'test_helper'
+
+class SmartProxyReferenceTest < ActiveSupport::TestCase
+  class ProxyReferrer
+    include SmartProxyReference
```

Comment:
I think this is not the intended usage. Should be changed to `SmartProxyHostExtensions`.